vispy/visuals/_scalable_textures.py

```diff
@@ -114,7 +114,19 @@ def set_clim(self, clim):
     @property
     def clim_normalized(self):
         """Normalize current clims to match texture data inside the shader.
+
+        If data is scaled on the CPU then the texture data will be in the range
+        0-1 in the _build_texture() method. Inside the fragment shader the
+        final contrast adjustment will be applied based on this normalized
+        ``clim``.
+
         """
+        if isinstance(self.clim, str) and self.clim == "auto":
+            raise RuntimeError("Can't return normalized color limits until "
+                               "data has been set. Call 'scale_and_set_data' "
+                               "first.")
```

Comment:
Maybe include the word "auto" in this error message to make it clear that it's related to auto-clim?